### PR TITLE
Minor fix

### PR DESCRIPTION
**Pull request by cyyever**
#### Why we need this PR?
*Pick one, remove the rest*
- Refactoring to improve *performance and fix some errors*

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *[ refactor code to improve performance and fix some errors ]*

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

File: dali/core/CMakeLists.txt [hunk 1]
```diff
@@ -31,10 +31,11 @@ endforeach(incl_dir)
 separate_arguments(DEFAULT_COMPILER_INCLUDE UNIX_COMMAND  "${DEFAULT_COMPILER_INCLUDE}")
 
 if (NOT LINK_DRIVER)
+  find_package (Python COMPONENTS Interpreter)
   set(CUDA_GENERATED_STUB "${CMAKE_CURRENT_BINARY_DIR}/dynlink_cuda_gen.cc")
   add_custom_command(
       OUTPUT ${CUDA_GENERATED_STUB}
-      COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/../../tools/stub_generator/stub_codegen.py --unique_prefix=Cuda --
+      COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/../../tools/stub_generator/stub_codegen.py --unique_prefix=Cuda --
```
Comment (mzient) [inline dali/core/CMakeLists.txt:38]: @JanuszL Are we sure it won't break anything in our build system?

File: dali/operators/reader/nvdecoder/cuvideodecoder.cc [hunk 2]
```diff
@@ -69,7 +69,7 @@ const char* GetVideoChromaFormatString(cudaVideoChromaFormat eChromaFormat) {
         { cudaVideoChromaFormat_444,        "YUV 444"              },
     };
 
-    if (static_cast<size_t>(eChromaFormat)
```
Comment (mzient) [inline dali/operators/reader/nvdecoder/cuvideodecoder.cc:72]: Please revert - the underlying type of an `enum` is implementation-defined and might be signed.

File: dali/core/CMakeLists.txt (hunk 1 above)
Comment (JanuszL) [reply dali/core/CMakeLists.txt:38]: It executes python that is visible in the shell. I guess `Python_EXECUTABLE` is the python that CMake can find, and probably available in the shell.
I think it should be fine.

**Comment by JanuszL** [comment]
!build

**Review by JanuszL — COMMENTED** [review]
I see that some of `test_operator_reader_shuffling` tests fails.
I guess it may be caused by `file_lable_loader.cc` changes, but I'm not sure.

File: dali/operators/reader/loader/file_label_loader.cc [hunk 3]
```diff
@@ -28,7 +28,7 @@ void FileLabelLoader::PrepareEmpty(ImageLabelWrapper &image_label) {
 }
 
 void FileLabelLoader::ReadSample(ImageLabelWrapper &image_label) {
-  auto image_pair = image_label_pairs_[current_index_++];
+  const auto &image_pair = image_label_pairs_[current_index_++];
```
Comment (mzient) [inline dali/operators/reader/loader/file_label_loader.cc:31]: A test fails - I guess this copy was intentional.

File: dali/core/CMakeLists.txt (hunk 1 above)
Comment (cyyever) [reply dali/core/CMakeLists.txt:38]: In some systems, there is no python but python3...

**Review by mzient — APPROVED** [review]

**Review by JanuszL — APPROVED** [review]

**Comment by JanuszL** [comment]
@cyyever - thank you for your contribution.

**Pull request merged**